### DISCUSSION TITLE
Quais ferramentas geralmente são usadas para cortar fotos quando vamos realizar upload?

### DISCUSSION THREAD

**Discussion by renedocket** [frontendbr/forum#1514] [Perguntas e Respostas]
Olá! Bom dia, pessoal! 

Bom, eu gostaria de saber se vocês já tiveram que tratar as imagens ao realizar as upload da mesma?

Um exemplo prático, seria definir um tamanho limite que a foto deve ter, e o usuário enquadra a foto como ele desejar, que consequentemente seria cortada ao realizar o upload no tamanho que eu determinei.

Pesquisei e testei algumas possibilidades como o Croppie, Croppic, Cropper.js, todas elas são com Jquery. 

Pessoal, vocês já tiverem esse desafio e quais ferramentas vocês usaram?

Obrigado desde já! :D

**Comment by yuritoledo** [comment]
Olá! :)

Não sei para qual stack você está procurando, mas eu uso esse e recomendo, para React: https://github.com/roadmanfong/react-cropper

Se eu não me engano, é baseado nesse Cropper.js, sendo a versão React da mesma.

Boa sorte ;)

**Comment by renedocket** [comment]
Opa, yuritoledo! Que legal, boa dica! No caso seria para JavaScript Vanilha. Não sei se tem algo do tipo. Mas obrigado! :)

**Comment by kivervinicius** [comment]
@renedocket  o proprio cropper.js agora tem versão vanilla.

**Comment by salatielsql** [comment]
Já viu esse? https://jamesooi.design/Croppr.js/

**Comment by renedocket** [comment]
Então pessoal, consegui testar o Cropper.js, achei bem maneiro! O problema que não consigo cortar as imagens sem precisar de back-end. Você sabem se isso é possível, fazer isso depender de backend?
Ou se existe algum coisa em JS Vanilla que nos possibilita essa ação?

**Comment by yuritoledo** [comment]
@renedocket se estive usando React, você consegue cortar a imagem sem precisar de backend, com o [cropper-react](https://www.npmjs.com/package/react-cropper)

**Comment by eliseumds** [comment]
Eu uso [CropperJS](https://github.com/fengyuanchen/cropperjs) também, excelente.
